test/support/generator/upload-file-in-deal-folder-generator.ts

```diff
@@ -0,0 +1,213 @@
+      _response: null,
+    };
+
+    const getSharepointSiteIdPath = `sites/${ENVIRONMENT_VARIABLES.SHAREPOINT_MAIN_SITE_NAME}.sharepoint.com:/sites/${values.ukefSiteId}`;
+
+    const getSharepointSiteIdResponse = {
+      id: values.sharepointSiteId,
+    };
+
+    const getDriveIdPath = `${getSharepointSiteIdPath}:/drives`;
+
+    const getDriveIdResponse = {
+      value: [
+        {
+          name: 'Case Library',
+          id: values.driveId,
+        },
+      ],
+    };
+
+    const fileDestinationPath = `${values.buyerName}/D ${values.dealId}`;
+
+    const urlToCreateUploadSession = `${ENVIRONMENT_VARIABLES.SHAREPOINT_BASE_URL}/sites/${values.sharepointSiteId}/drives/${values.driveId}/root:/${fileDestinationPath}/${values.fileName}:/createUploadSession`;
+
+    const uploadSessionHeaders = {
+      item: {
+        '@microsoft.graph.conflictBehavior': 'fail',
+      },
+    };
+
+    const getUploadSessionArgs: [
+      string,
+      {
+        item: {
+          '@microsoft.graph.conflictBehavior': string;
+        };
+      },
+    ] = [urlToCreateUploadSession, uploadSessionHeaders];
+
+    const uploadSession: LargeFileUploadSession = {
+      url: values.uploadSessionUrl,
+      expiry: values.uploadSessionExpiry,
+    };
+
+    const uploadTaskOptions: LargeFileUploadTaskOptions = { rangeSize: fileSizeInBytes };
+
+    const getUploadTaskArgs: [string, number, LargeFileUploadSession, LargeFileUploadTaskOptions] = [
+      values.fileName,
+      fileSizeInBytes,
+      uploadSession,
+      uploadTaskOptions,
+    ];
+
+    const getListIdPath = `${getSharepointSiteIdPath}:/lists`;
+
+    const getListIdResponse = { value: [{ name: 'CaseLibrary', id: values.listId }] };
+
+    const getItemIdPath = `${getSharepointSiteIdPath}:/lists/${values.listId}/items`;
+
+    const encodedBuyerName = encodeURIComponent(values.buyerName);
+    const encodedDealId = encodeURIComponent(values.dealId);
+    const encodedFileDestinationPath = `${encodedBuyerName}/${encodeURIComponent('D ')}${encodedDealId}`;
+    const encodedFileName = encodeURIComponent(values.fileName);
+
+    const itemWebUrl = `https://${ENVIRONMENT_VARIABLES.SHAREPOINT_MAIN_SITE_NAME}.sharepoint.com/sites/${values.ukefSiteId}/CaseLibrary/${encodedFileDestinationPath}/${encodedFileName}`;
+
+    const getItemIdResponse = { value: [{ webUrl: itemWebUrl, id: values.itemId }] };
+
+    const updateFileInfoPath = `${getItemIdPath}/${values.itemId}/fields`;
+
+    const { documentTitle, documentTypeId } = new DocumentTypeMapper({
+      estoreDocumentTypeIdApplication: ENVIRONMENT_VARIABLES.SHAREPOINT_ESTORE_DOCUMENT_TYPE_ID_APPLICATION,
+      estoreDocumentTypeIdFinancialStatement: ENVIRONMENT_VARIABLES.SHAREPOINT_ESTORE_DOCUMENT_TYPE_ID_FINANCIAL_STATEMENT,
+      estoreDocumentTypeIdBusinessInformation: ENVIRONMENT_VARIABLES.SHAREPOINT_ESTORE_DOCUMENT_TYPE_ID_BUSINESS_INFORMATION,
+    }).mapDocumentTypeToTitleAndTypeId(values.documentType);
```

Comment:
Using the whole mapper feels like a lot of logic to have in the generator. It means the tests can't test the mapper logic at all. Maybe it'd be better to just pick 1 document type for the generator, and the test needs to set up the data for the other enum values itself?

Reply:
Happy for you to resolve this in your next PR though

Reply:
Will resolve in next PR